helium-lib/src/dao.rs

```diff
@@ -186,29 +183,52 @@ impl SubDao {
     }
 
     pub fn rewardable_entity_config_key(&self) -> Pubkey {
-        let suffix = match self {
-            Self::Iot => b"IOT".as_ref(),
-            Self::Mobile => b"MOBILE".as_ref(),
-        };
-        let (key, _) = Pubkey::find_program_address(
-            &[b"rewardable_entity_config", self.key().as_ref(), suffix],
-            &helium_entity_manager::id(),
-        );
-        key
+        let sub_dao = self.key();
+        match self {
+            Self::Iot => {
+                Pubkey::find_program_address(
+                    &[
+                        b"rewardable_entity_config",
+                        &sub_dao.as_ref(),
+                        "IOT".as_bytes(),
```

Comment:
why is this change here? must more concise in the previous version with no repitition?